facade/src/main/java/org/jboss/pnc/facade/providers/DeliverableAnalyzerReportProviderImpl.java

```diff
@@ -98,6 +126,88 @@ public Page<AnalyzedArtifact> getAnalyzedArtifacts(
                 analyzedArtifacts);
     }
 
+    @RolesAllowed(SYSTEM_USER)
+    @Override
+    public void addLabel(String id, DeliverableAnalyzerReportLabelRequest request) {
+        Base32LongID reportId = transformToEntityId(id);
+        DeliverableAnalyzerReport report = deliverableAnalyzerReportRepository.queryById(reportId);
+        DeliverableAnalyzerLabelEntry labelHistoryEntry = DeliverableAnalyzerLabelEntry.builder()
+                .report(report)
+                .changeOrder(deliverableAnalyzerLabelEntryRepository.getLatestChangeOrderOfReport(report.getId()))
+                .entryTime(Date.from(Instant.now()))
+                .user(userService.currentUser())
+                .reason(request.getReason())
+                .label(request.getLabel())
+                .change(LabelOperation.ADDED)
+                .build();
+
+        labelModifier.addLabelToActiveLabelsAndModifyLabelHistory(
+                reportId,
+                request.getLabel(),
+                report.getLabels(),
+                labelHistoryEntry);
+    }
+
+    @RolesAllowed(SYSTEM_USER)
```

Comment:
I think we want to allow all users to manipulate the labels.

Reply:
[94be71d](https://github.com/project-ncl/pnc/pull/3840/commits/94be71d1c617873573ff88072d45893d197c41ad)